scripts/reinforcement_learning/ray/task_runner.py

```diff
@@ -0,0 +1,176 @@
+import yaml
+import ray
+import sys
+import argparse
+import subprocess
+import threading
+from enum import Enum
+
+"""
+This script dispatches one or more user-defined Python tasks to workers in a Ray cluster.
+Each task, with its resource requirements and execution parameters, is described in a YAML configuration file.
+You may specify the desired number of CPUs, GPUs, and memory allocation for each task in the config file.
+
+Key features:
+- Flexible resource management per task via config fields (`num_gpus`, `num_cpus`, `memory`).
+- Real-time output streaming (stdout/stderr) for each task.
+- Parallel execution of multiple tasks across cluster resources.
+
+Tasks are distributed and scheduled according to Ray’s built-in resource manager.
+
+Typical usage:
+---------------
+
+.. code-block:: bash
+
+    # Print help and argument details:
+    python task_runner.py -h
+
+    # Submit tasks defined in a YAML file to the Ray cluster (auto-detects Ray head address):
+    python task_runner.py --task_cfg /path/to/tasks.yaml
+
+YAML configuration example:
+---------------------------
+.. code-block:: yaml
+    pip: ["xxx"]
+    py_modules: ["my_package/my_package"]
+    tasks:
+      - name: "task1"
+        py_args: "-m torch.distributed.run --nnodes=1 ..."
+        num_gpus: 2
+        num_cpus: 10
+        memory: 10737418240
+      - name: "task2"
+        py_args: "script.py --option arg"
+        num_gpus: 0
+        num_cpus: 1
+        memory: 10*1024*1024*1024
+
+- `pip`: List of pip packages to install.
+- `py_args`: Arguments passed to the Python executable for this task.
+- `num_gpus`, `num_cpus`: Number of GPUs/CPUs to allocate. Can be integer or a string like `"2*2"`.
+- `memory`: Amount of memory (bytes) to allocate. Can be integer or a string like `"10*1024*1024*1024"`.
+
+To stop all tasks early, press Ctrl+C; the script will cancel all running Ray tasks.
+"""
+
+class OutputType(str, Enum):
+    STDOUT = "stdout"
+    STDERR = "stderr"
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Run tasks from a YAML config file.")
+    parser.add_argument("--task_cfg", type=str, required=True, help="Path to the YAML task file.")
+    parser.add_argument("--ray_address", type=str, default="auto", help="the Ray address.")
+    return parser.parse_args()
+    
+@ray.remote
+def task_wrapper(task):
```

Comment:
I think you almost def may not want to handle the stdout and stderr logic here. We already have a piece of code that is similar to this, and it is already kinda brittle; Please see https://github.com/isaac-sim/IsaacLab/pull/2346

Reply:
Check out https://github.com/binw666/IsaacLab/blob/752be19bade88c1f1e2a06a1ca6519baafbba216/scripts/reinforcement_learning/ray/util.py#L115

Reply:
and https://github.com/binw666/IsaacLab/blob/752be19bade88c1f1e2a06a1ca6519baafbba216/scripts/reinforcement_learning/ray/util.py#L100